ontolearner/learner/taxonomy_discovery/skhnlp.py

```diff
@@ -0,0 +1,761 @@
+    TrainingArguments,
+)
+
+from ...base import AutoLearner, AutoPrompt
+from ...utils import taxonomy_split, train_test_split as ontology_split
+from ...data_structure import OntologyData, TaxonomicRelation
+
+class SKHNLPTaxonomyPrompts(AutoPrompt):
+    """Builds the 7 taxonomy prompts used during fine-tuning / inference."""
+    def __init__(self) -> None:
+        super().__init__(prompt_template="{parent} is the superclass of {child}. This statement is [MASK].")
+        self.templates: List[str] = [
+            "{parent} is the superclass of {child}. This statement is [MASK].",
+            "{child} is a subclass of {parent}. This statement is [MASK].",
+            "{parent} is the parent class of {child}. This statement is [MASK].",
+            "{child} is a child class of {parent}. This statement is [MASK].",
+            "{parent} is a supertype of {child}. This statement is [MASK].",
+            "{child} is a subtype of {parent}. This statement is [MASK].",
+            "{parent} is an ancestor class of {child}. This statement is [MASK].",
+        ]
+
+    def make(self, parent: str, child: str, template_idx: int) -> str:
+        return self.templates[template_idx].format(parent=parent, child=child)
+
+
+class SKHNLPSequentialFTLearner(AutoLearner):
+    """
+    BERT-based classifier for taxonomy discovery.
+
+    With OntologyData:
+      * TRAIN: ontology-aware split; create balanced train/eval with negatives.
+      * PREDICT/TEST: notebook-style parent selection -> list[{'parent', 'child'}].
+
+    With DataFrame/list:
+      * TRAIN: taxonomy_split + negatives; build prompts and fine-tune.
+      * PREDICT/TEST: pairwise binary classification (returns label + score).
+    """
+
+    # Fixed constants defining data split size and the proportional mix of
+    # negative sample types (reversed vs. manipulated) for balancing.
+    _EVAL_FRACTION: float = 0.16
+    _NEG_RATIO_REVERSED: float = 1/3
+    _NEG_RATIO_MANIPULATED: float = 2/3
+
+    def __init__(
+        self,
+        # core
+        model_name: str = "bert-large-uncased",
+        n_prompts: int = 7,
+        random_state: int = 1403,
+        device: Optional[str] = None,  # "cuda" | "cpu" | None (auto)
+
+        # ---- expose TrainingArguments as individual user-defined args ----
+        output_dir: str = "./results/",
+        num_train_epochs: int = 1,
+        per_device_train_batch_size: int = 4,
+        per_device_eval_batch_size: int = 4,
+        warmup_steps: int = 500,
+        weight_decay: float = 0.01,
+        logging_dir: str = "./logs/",
+        logging_steps: int = 50,
+        eval_strategy: str = "epoch",
+        save_strategy: str = "epoch",
+        load_best_model_at_end: bool = True,
+    ) -> None:
+        super().__init__()
+        self.model_name = model_name
+        self.n_prompts = n_prompts
+        self.random_state = random_state
+        self.device = device or ("cuda" if torch.cuda.is_available() else "cpu")
```

Comment:
remove the `remove ` ("cuda" if torch.cuda.is_available() else "cpu")` it should ` and add default value of `cpu` for `device` in arguments.